### DISCUSSION TITLE
AttributeError: 'RobertaModel' object has no attribute 'layer_idx'

### DISCUSSION THREAD

**Discussion by tangshiyi341914** [adapter-hub/adapters#628] [Q&A] [answered]
Hello,

I am a student of computer science and I am working on AdapterDrop.

When I ran the Adapter_Drop_Training.ipynb step by step,I got an error:

transformers/trainer.py:1860, in Trainer._inner_training_loop(self, batch_size, args, resume_from_checkpoint, trial, ignore_keys_for_eval)
self.control = self.callback_handler.on_step_begin(args, self.state, self.control)
with self.accelerator.accumulate(model):
tr_loss_step = self.training_step(model, inputs)
 if (
     args.logging_nan_inf_filter
     and not is_torch_tpu_available()

    and (torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step))
...
   if name in modules:
   return modules[name]
   raise AttributeError(f"'{type(self).__name__}' object has no attribute '{name}'")
AttributeError: 'RobertaModel' object has no attribute 'layer_idx'

I don't know how to solve it,could you help me？

**Comment by calpt** [comment, marked as answer]
Hey @tangshiyi341914, thanks for reporting this! This looks like an issue in our library, I've converted this into an issue here: #633 and will look into fixing it.

**Reply by tangshiyi341914** [reply]
Thanks a lot！

**Reply by calpt** [reply]
This should be fixed on our main branch now. Please mark as solved if this fixes your issue. Thanks!

**Reply by tangshiyi341914** [reply]
Thanks for your watching！
